### ISSUE TITLE
Concepts - Output

### ISSUE THREAD

**Issue by bebraw** [webpack/webpack.js.org#10]
[Stub](https://github.com/webpack/webpack.io/blob/develop/content/concepts/output.md).

Feel free to comment here if you have ideas on what this guide should cover. Link to potential resources too.

**Comment by michael-ciniawsky** [comment]
@bebraw @TheLarkInn Working on it, loaders aswell, when the basic document structure stands a open PR for review

**Comment by michael-ciniawsky** [comment]
# Output

[Intro]

## Usage
[Description]

**webpack.config.js** 

[Example (basic)]

## Options
  - path 
  - publicPath
  - library (maybe separated ?)
  - libraryTarget (maybe separated ?)
  - filename
  - chunkFilename
  - sourceMapFilename
  - jsonpFunction (maybe advanced ?)

## Advanced Options
### devtool 
   - devtoolLineToLine
   - devtoolModuleFilenameTemplate
   - devtoolFallbackModuleFilenameTemplate

### hotUpdate
   - hotUpdateFunction
   - hotUpdateChunkFilename
   - hotUpdateMainFilename

### No section assign yet ( ? Ideas welcome :))
   - pathinfo
   - sourcePrefix
   - umdNamedDefine
   - crossOriginLoading

Are any of this options deprecated or should be avoided ? Roughly structure ok so far? :)

**Comment by gregvenech** [comment]
@michael-ciniawsky structure looks good to me 👍 , just a heads up though: you might want to take a look #157, as all these options you're listing already have at least a start there. See the comments, we're currently trying to figure out how to divvy each section out. We're you planning on documenting each on this page or just linking to it?

@bebraw aside from the overlap here we should probably look at #156 as some of that is already done or started #157 as well.

**Comment by gregvenech** [comment]
The structure @michael-ciniawsky laid out above looks like it could work very well for the split up we've been talking about #157.

**Comment by bebraw** [comment]
Yeah, there's some definite overlap. The point is that you can go into great detail in this document if we agree that the configuration page is an overview (quick reference).

**Comment by gregvenech** [comment]
@bebraw but then we should just port all those detailed options from #157 to the individual Concepts pages like this one. I'm just going to go ahead and do that and merge #157 before any more overlap is created. If we want to rename the top-level nav items down the road we can. That work?

**Comment by bebraw** [comment]
Yeah, merge is ok I think.

**Comment by michael-ciniawsky** [comment]
> We're you planning on documenting each on this page or just linking to it?

@gregvenech 

Depending on how you guys plan to do it :), it's the intended structure + document each on the page, but yes I noticed early that it would get 'a smell' of the current docs quickly. The organisation/layout must be different somehow. 

Is it possible to display the advanced options + example programmatically on the concept pages e.g via user selection

```html
<select>
  <option value="beginner">Beginner</option>
  <option value="advanced">Advanced</option>
</select>
```
which hides all advanced options + example on ever page when beginner(default) is selected ? This goes in the direction like you and sokra discussed hiding advanced content in  ```<details></details>```, but maybe a diff between beginner and advanced in terms of two different 'walkthroughs' would make docs usage easier to follow at the beginning, manual clicking on each respective page's details could be enabled nevertheless for folks wanting to revisit a specified section. As @bebraw stated out the right amout of detail is the tricky and important part, speaking of **output** for example , chunking and sourcemaps + OccurenceOrderPlugin and friends are 'advanced' beginner contents imo.

When I started, the 'basic' example + options are straightforward to document, there are not much ambiguities, but the advanced options will cause confusion for beginners since it's unlikley they have dig into webpack internals (compiler, code gen) at this point of their journey.

**Comment by gregvenech** [comment]
@michael-ciniawsky 

Yea I do like the structure you described but it could get a bit lengthy having the full documentation for all of those options there. I really like that you tied in the other options that relate to `output` but aren't nested as options under output. I definitely think these ties should at least be linked to from the various configuration, concept, and api pages.

> Is it possible to display the advanced options + example programmatically on the concept pages e.g via user selection...

Anything's possible 😁  ! Personally I think we can get a lot of that breakdown just by splitting up pages, organizing them (#93), and linking to them -- creating an intuitive flow. Something like what you described could be useful but it would require some tweaks to the markdown parser or Page component to support it.

I'm working on splitting up #157 into multiple pages under a new **Configuration** section. I'm hoping to have it done and merged tonight. Maybe it's worth us quickly regrouping a little bit after and deciding on a distinct purpose for the pages within each section, i.e. **Configuration**, **Concepts**, **API**, etc., before diving in too much more. That way we can have a more consistent "game plan" when approaching new pages. @bebraw what do you think?

**Comment by bebraw** [comment]
@gregvenech Yeah, we can sync after #157 is in a good shape and split up. I wouldn't get too stuck on the structure. It's easier to make judgment calls once there's something visible. I don't mind editing things into shape. 👍

**Comment by gregvenech** [comment]
@bebraw perfect.

**Comment by michael-ciniawsky** [comment]
@gregvenech @bebraw  👍 i wait until the split up is done, preparing contents meanwhile.

**Comment by gregvenech** [comment]
@michael-ciniawsky it's been split and merged. @sokra already started revising the new **Configuration** section in #180. So I guess anything on how to use specific `output` options would go in `/configuration/output` while the why and what is `output` would go in `/concepts/output`... would you agree @bebraw, @TheLarkInn?

**Comment by bebraw** [comment]
Sounds good to me.

**Comment by michael-ciniawsky** [comment]
@gregvenech 👍

Off topic but are there any notes besides the release note about the changes in modules (loaders -> rules) etc.? So I can update it and commit (concepts/modules, concepts/loaders)

**Comment by bebraw** [comment]
@michael-ciniawsky Two sources for now: [migration guide](http://webpack.js.org/how-to/upgrade-from-webpack-1/), [gist](https://gist.github.com/sokra/27b24881210b56bbaff7).

**Comment by TheLarkInn** [comment]
Any update on this? Can help if anyone is stuck.

**Comment by bebraw** [comment]
[Online already](https://webpack.js.org/concepts/output/). Just forgot to close this one.
